### PR TITLE
Add HPL Docsite

### PR DESCRIPTION
**Pull request by Perronef5**
## CHANGES 💻

- Add HPL Docsite
- Auto Generate IDL Docs
- Add Getting Started Docs
- Add Overview Docs
- Add how to interact with lazy distributor docs
- Also added a banner in the helium docs overview page that links to this website here https://github.com/helium/docs/pull/1818

## DEMO 📷

https://github.com/helium/helium-program-library/assets/19807431/74231348-efb5-457f-8070-76305b3fd6ac

### CODE REVIEW

**Comment by ChewingGlass** [comment]
@Perronef5 this broke the builds

File: packages/docsite/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Tailwind UI License
```
Comment (ChewingGlass) [inline packages/docsite/LICENSE.md:1]: What? Why

Comment (Perronef5) [reply packages/docsite/LICENSE.md:1]: Hmmm let me remove

File: packages/docsite/src/pages/docs/api/nft-voter-sdk.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# Nft Voter SDK
+
+{% callout title="Quick tip" %}
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/api/nft-voter-sdk.md:3]: This isn't a package in hpl?

File: packages/docsite/src/pages/docs/api/no-emit-sdk.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# No Emit SDK
+
+{% callout title="Quick tip" %}
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/api/no-emit-sdk.md:3]: what?

File: packages/docsite/src/pages/docs/api/token-voter-sdk.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
+# Token Voter SDK
+
+{% callout title="Quick tip" %}
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/api/token-voter-sdk.md:3]: not in hpl

File: packages/docsite/src/pages/docs/api/vote-hook-interface-sdk.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Vote Hook Interface SDK
+
+{% callout title="Quick tip" %}
+If you are looking for a quick start guide, check out the [Getting Started](/docs/learn/getting_started) guide. We also have some react examples [here](/docs/learn/react).
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/api/vote-hook-interface-sdk.md:4]: not in hpl

File: packages/docsite/src/pages/docs/installation.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+---
+title: Installing Helium Program Library
+description: Learn how to install Helium Program Library
+---
+
+Modular Governance is a set of smart contracts that can be used to create a governance system for your project. It is designed to be modular and flexible so that it can be used for a wide variety of use cases.
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/installation.md:6]: Fix

File: packages/docsite/src/pages/docs/learn/getting_started.md [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+Helium Program Library lets you build on the Helium Network.
+
+Looking to learn more about the Helium Program Library? You're in the right place!
+
+## Initializing the SDK
+
+Let's get started by installing the sdks
+
+```shell
+yarn add @helium/account-fetch-cache @helium/account-fetch-cache-hooks @helium/address @helium/crypto-react-native @helium/currency @helium/currency-utils @helium/data-credits-sdk @helium/distributor-oracle @helium/helium-entity-manager-sdk @helium/helium-react-hooks @helium/helium-sub-daos-sdk @helium/hotspot-utils @helium/http @helium/idls @helium/lazy-distributor-sdk @helium/spl-utils
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/learn/getting_started.md:17]: This isn't all the packages. Why this set of packages?

File: packages/docsite/src/pages/docs/learn/getting_started.md [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+            hotspot.pendingRewards &&
+            hotspot.pendingRewards[mint.toString()] &&
+            new BN(hotspot.pendingRewards[mint.toString()]).gt(new BN(0))
+          )
+            acc += 1
+        })
+        return acc
+      }, 0)
+      dispatch(
+        solanaSlice.actions.setPaymentProgress({
```
Comment (ChewingGlass) [inline packages/docsite/src/pages/docs/learn/getting_started.md:178]: Needs better example code. This is mixed in with nasty redux logic.

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

File: packages/docsite/src/pages/docs/learn/getting_started.md (hunk 7 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/learn/getting_started.md:17]: I removed and added some basic ones

File: packages/docsite/src/pages/docs/api/no-emit-sdk.md (hunk 3 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/api/no-emit-sdk.md:3]: Removed

File: packages/docsite/src/pages/docs/api/token-voter-sdk.md (hunk 4 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/api/token-voter-sdk.md:3]: Removed

File: packages/docsite/src/pages/docs/api/nft-voter-sdk.md (hunk 2 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/api/nft-voter-sdk.md:3]: Removed

File: packages/docsite/src/pages/docs/learn/getting_started.md (hunk 8 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/learn/getting_started.md:178]: There was only one set of leftover redux logic. Its been removed

**Review by ChewingGlass — APPROVED** [review]

File: packages/docsite/src/pages/docs/installation.md (hunk 6 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/installation.md]: Done

File: packages/docsite/src/pages/docs/api/vote-hook-interface-sdk.md (hunk 5 above)
Comment (Perronef5) [reply packages/docsite/src/pages/docs/api/vote-hook-interface-sdk.md]: Removed

**Comment by Perronef5** [comment]
DeWiCats

**Pull request merged**